### DISCUSSION TITLE
.NET Length UInt64, Span etc

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#71620] [General] [answered]
Since processors are 64 bit, why not use UInt64 by default on things like length of IEnumerable

**Comment by Clockwork-Muse** [comment]
1. (Massive) Breaking change
 2. There's some debate over whether sizes should be unsigned at all (due to conversion rules, among other things).
 3. In any real-world application it's unlikely to be practical to iterate over that many elements in an `IEnumerable`.  Getting a 2GB+ contiguous span of memory (for an array or similar) is barely supported.

**Comment by huoyaoyuan** [comment, marked as answer]
Se also the discussion in #12221

**Comment by Xyncgas** [comment]
Thanks for the infos
